pkg/cassandra/encryption.go

```diff
@@ -0,0 +1,217 @@
+package cassandra
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/encryption"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	StoresMountPath = "/mnt"
+)
+
+// Sets up encryption in the datacenter config template.
+// The keystore and truststore config maps are mounted into the datacenter pod and the secrets are read to be set in the datacenter config template.
+func HandleEncryptionOptions(template *DatacenterConfig) error {
+	if ClientEncryptionEnabled(template) {
+		if err := checkMandatoryEncryptionFields(template.ClientEncryptionStores); err != nil {
+			return err
+		} else {
+			// Create the volume and mount for the keystore
+			addVolumesForEncryption(template, "client", *template.ClientEncryptionStores)
+		}
+	}
+
+	if ServerEncryptionEnabled(template) {
+		if err := checkMandatoryEncryptionFields(template.ServerEncryptionStores); err != nil {
+			return err
+		} else {
+			// Create the volume and mount for the keystore
+			addVolumesForEncryption(template, "server", *template.ServerEncryptionStores)
+		}
+	}
+	return nil
+}
+
+func addVolumesForEncryption(template *DatacenterConfig, storeType string, encryptionStores encryption.EncryptionStores) {
+	// Initialize the volume array if it doesn't exist yet
+	if template.PodTemplateSpec.Spec.Volumes == nil {
+		template.PodTemplateSpec.Spec.Volumes = make([]corev1.Volume, 0)
+	}
+
+	volumes := EncryptionVolumes(storeType, encryptionStores)
+
+	for _, volume := range volumes {
+		indexKey, foundKey := FindVolume(template.PodTemplateSpec, volume.Name)
+		AddOrUpdateVolume(template, &volume, indexKey, foundKey)
+	}
+
+	// Find the cassandra container and add the volume mounts for both stores
+	addEncryptionMountToCassandra(template, volumes["keystore"], volumes["truststore"], storeType)
+}
+
+func EncryptionVolumes(storeType string, encryptionStores encryption.EncryptionStores) map[string]corev1.Volume {
```

Comment:
I see a `storeType` parameter in many functions but it seems it can only be `client` or `server`. We coudl capture this with a dedicated type:

```go
type StoreType string

const (
	StoreTypeClient = StoreType("client")
	StoreTypeServer = StoreType("server")
)

func EncryptionVolumes(storeType StoreType, encryptionStores encryption.EncryptionStores) map[string]corev1.Volume {
...
```

Reply:
Done